### ISSUE TITLE
Support for Cloudfunctions 2nd gen

### ISSUE THREAD

**Issue by cmeury** [terraform-google-modules/terraform-google-scheduled-function#81]
### TL;DR

Is supporting the [2nd generation Cloud Functions](https://cloud.google.com/functions/docs/concepts/version-comparison#new-in-2nd-gen) in consideration for this module?

### Terraform Resources

* [cloudfunctions2_function](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions2_function)
* [cloudfunctions2_function_iam](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions2_function_iam)

### Additional information

* Added as GA to the `google` provider with version [4.34.0](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4340-august-29-2022)

**Comment by cmeury** [comment]
I think this would be a great addition? Is this the wrong place for the feature request?

**Comment by cmeury** [comment]
Nope, still shouldn't be stale.

**Comment by cmeury** [comment]
Still relevant.
